modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/WFSContentComplexFeatureCollection.java

```diff
@@ -99,6 +99,7 @@ public FeatureIterator<Feature> features() {
                 return new ComplexFeatureIteratorImpl(parser);
             } else {
                 ComplexGetFeatureResponse response = client.issueComplexRequest(request);
+                response.getParser().setFilter(filter);
                 return response.features();
```

Comment:
This is more of a question than a negative comment, but makes me wonder... if the request has not been sent out yet, why not merge the filter in the request and have the server deal with it, rather than having the parser do so, which I guess causes the filter to be applied client side? I guess it's just easier this way, or there are issues encoding the filters?